src/key.cpp

```diff
@@ -431,6 +420,21 @@ KeyPair::KeyPair(const CKey& key, const uint256* merkle_root)
     if (!success) ClearKeyPairData();
 }
 
+bool KeyPair::SignSchnorr(const uint256& hash, Span<unsigned char> sig, const uint256& aux) const
+{
+    assert(sig.size() == 64);
```

Comment:
`m_keypair->data()` can be a `nullptr`, should we return `false` early if it is?

Reply:
Good point. Since a KeyPair can be constructed and in an invalid state, I've moved the `if (!IsValid()) return false;` from `CKey::SignSchnorr` to here.